docs/design/2021-09-29-secure-bootstrap.md

```diff
@@ -0,0 +1,82 @@
+# TiDB Design Documents
+
+- Author(s): [morgo](http://github.com/morgo)
+- Discussion PR: https://github.com/pingcap/tidb/pull/28482
+- Tracking Issue: https://github.com/pingcap/tidb/issues/28481
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+Currently, TiDB will create a root user with no password and bind to `0.0.0.0/::`. This is a potential security issue if users do not have a firewall configured correctly, or if a rogue actor has local access.
+
+This document introduces the `--intialize-secure` and `--initialize-insecure` bootstrap options. The current behavior is equivalent to `--initialize-insecure`, but the intention is to change the default to secure **once** it is determined stable.
+
+## Motivation or Background
+
+The motivation for this change is to improve the default security of TiDB. The current default is insecure, and requires additional steps to be taken by users to secure TiDB.
+
+The design attempts to make the change with the least breakage possible. However, there will be some impact to usability.
+
+## Detailed Design
+
+The current bootstrap (to be referred to as `--initialize-insecure`) creates a root user as follows. This will remain unchanged:
+
+```sql
+CREATE USER 'root'@'%' IDENTIFIED WITH 'mysql_native_password' AS '' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK;
+GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
+```
+
+The `--initialize-secure` bootstrap will instead create a user based on the OS user. For example, with an OS user of `ubuntu`:
+
+```sql
+CREATE USER 'ubuntu'@'localhost' IDENTIFIED WITH 'auth_socket' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK;
+GRANT ALL PRIVILEGES ON *.* TO 'ubuntu'@'localhost' WITH GRANT OPTION;
+```
+
+After TiDB has bootstrapped, the user will be able to login and create additional users:
+
+```bash
+mysql -S /tmp/tidb.sock
+..
+mysql> CREATE USER 'root'@'%' IDENTIFIED BY 'securePassword';
+mysql> GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
+```
+
+### Required Supporting Features
+
+In order to add the `--initialize-secure` bootstrap option, the following supported features are required:
+
+1. Support for Socket Authentication (`auth_socket`) [PR in Review](https://github.com/pingcap/tidb/pull/27561)
+2. Stale socket files are automatically cleaned up on server start [PR Merged](https://github.com/pingcap/tidb/pull/27886)
+3. TiDB listens on both TCP **and** unix socket by default [PR in Review](https://github.com/pingcap/tidb/pull/28486)
+
+## Test Design
```

Comment:
If we use `ubuntu` instead of `root` we should verify if there are other things that might break because of this.

Might be good to test br, dumpling, lightning, etc to see if they allow UNIX socket connections (should not be a blocker, and likely already the case for MySQL support)

Reply:
Yes. Although in practice, this won't break if we modify `tiup` and Operator to use `--initialize-insecure` and generate a random password *before* we change the default bootstrap to secure (default change is not in scope yet for this proposal).